AmplifyPlugins/Storage/Sources/AWSS3StoragePlugin/AWSS3StoragePlugin+AsyncClientBehavior.swift

```diff
@@ -136,12 +136,12 @@ extension AWSS3StoragePlugin {
         options: StorageDownloadFileOperation.Request.Options? = nil
     ) -> StorageDownloadFileTask {
         let options = options ?? StorageDownloadFileRequest.Options()
-        let path = "" //TODO: resolve path
-        let request = StorageDownloadFileRequest(key: path, local: local, options: options)
+        let request = StorageDownloadFileRequest(key: "", local: local, options: options)
```

Comment:
I thought about adding path to `StorageDownloadFileRequest` but since the request is also being used in the deprecated path, how do you think we differentiate when to use `key` in the deprecated path and when to use `path` in the new path? Is there a boolean flag that determines whether to use key or path when we need to execute the transfer in the operation?